adapters/amx/amx_test.go

```diff
@@ -0,0 +1,335 @@
+
+	assert.Len(t, body.Imp, 1, "expecting 1 bid")
+
+	resps, errs := adapter.MakeBids(body, &adapters.RequestData{}, &adapters.ResponseData{
+		StatusCode: 200,
+		Body: []byte(`{
+			"id": "WQ5V2DWVTMNXABDD",
+			"seatbid": [{
+				"bid": [{
+					"id": "TEST",
+					"impid": "1",
+					"price": 10.0,
+					"adid": "1",
+					"adm": "<?xml version=\"1.0\" encoding=\"UTF-8\" ?><VAST version=\"2.0\"><Ad id=\"128a6.44d74.46b3\"><InLine><Error><![CDATA[http://example.net/hbx/verr?e=]]></Error><Impression><![CDATA[http://example.net/hbx/vimp?lid=test&aid=testapp]]></Impression><Creatives><Creative sequence=\"1\"><Linear><Duration>00:00:15</Duration><TrackingEvents><Tracking event=\"firstQuartile\"><![CDATA[https://example.com?event=first_quartile]]></Tracking></TrackingEvents><VideoClicks><ClickThrough><![CDATA[http://example.com]]></ClickThrough></VideoClicks><MediaFiles><MediaFile delivery=\"progressive\" width=\"16\" height=\"9\" type=\"video/mp4\" bitrate=\"800\"><![CDATA[https://example.com/media.mp4]]></MediaFile></MediaFiles></Linear></Creative></Creatives></InLine></Ad></VAST>",
+					"adomain": ["amxrtb.com"],
+					"iurl": "https://assets.a-mo.net/300x250.v2.png",
+					"cid": "1",
+					"crid": "1",
+					"h": 600,
+					"w": 300,
+					"ext": {
+						"himp": ["https://example.com/imp-tracker/pixel.gif?param=1&param2=2"],
+						"startdelay": 0
+					}
+				}]
+			}],
+			"cur": "USD"
+		}`),
+	})
+
+	assert.Empty(t, errs, "unexpected errors in response")
+	assert.Len(t, resps.Bids, 1, "there should only be 1 bid")
+
+	// it should be a video bid
+	assert.Equal(t, openrtb_ext.BidTypeVideo, resps.Bids[0].BidType, "the bid should be video type")
+}
+
+func TestUserEidsOnly(t *testing.T) {
+	var w, h int = 300, 250
+
+	var width, height uint64 = uint64(w), uint64(h)
+
+	adapter := NewAMXBidder(amxTestEndpoint)
+	imp1 := openrtb.Imp{
+		ID:  "imp1",
+		Ext: json.RawMessage(defaultImpExt),
+		Banner: &openrtb.Banner{
+			W: &width,
+			H: &height,
+			Format: []openrtb.Format{
+				{W: 300, H: 250},
+			},
+		}}
+
+	inputRequest := openrtb.BidRequest{
+		Imp: []openrtb.Imp{imp1, imp1, imp1},
+		Site: &openrtb.Site{
+			Publisher: &openrtb.Publisher{
+				ID: "10007",
+			},
+		},
+		User: &openrtb.User{Ext: json.RawMessage(
+			`{
+				"eids": [{
+				"source": "adserver.org",
+					"uids": [{
+						"id": "111111111111",
+						"ext": {
+							"rtiPartner": "TDID"
+						}
+					}]
+				},{
+					"source": "example.buid",
+					"uids": [{
+						"id": "123456"
+					}]
+				}]
+			}`)},
+		ID: "1234",
+	}
+
+	actualAdapterRequests, _ := adapter.MakeRequests(&inputRequest, &adapters.ExtraRequestInfo{})
+	assert.Len(t, actualAdapterRequests, 1, "there should be 1 request")
+
+	body := getRequestBody(actualAdapterRequests)
+	assert.NotNil(t, body, "the generated OpenRTB request is not valid JSON")
+
+	userExt := getUserExt(body.User)
+	assert.NotNil(t, userExt, "the generated user.ext is invalid")
+
+	assert.Len(t, userExt.Eids, 2)
```

Comment:
Not exactly sure what exactly this test is testing because I don't see any modifications around user ext in your `MakeRequests` function. This to me looks more like making sure that user exts are not getting modified by `MakeRequests`. Is that correct? 

Also, I'd suggest adding these tests in this file as part of the JSON tests rather than unit tests as that will help remove a lot of setup for your tests which is covered by `RunJSONBidderTest` and will also provide consistency in tests with other adapters. In fact some of these test cases are already covered with the JSON tests.